src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -426,5 +432,929 @@ public void EmptyRecord()
                 Diagnostic(ErrorCode.ERR_BadRecordDeclaration, "()").WithLocation(2, 13)
             );
         }
+
+        [Fact]
+        public void WithExpr1()
+        {
+            var src = @"
+data class C(int X)
+{
+    public static void Main()
+    {
+        var c = new C(0);
+        _ = Main() with { };
+        _ = default with { };
+        _ = null with { };
+    }
+}";
+            var comp = CreateCompilation(src);
+            comp.VerifyDiagnostics(
+                // (7,13): error CS8802: The receiver of a `with` expression must have a valid non-void type.
+                //         _ = Main() with { };
+                Diagnostic(ErrorCode.ERR_InvalidWithReceiverType, "Main()").WithLocation(7, 13),
+                // (8,13): error CS8716: There is no target type for the default literal.
+                //         _ = default with { };
+                Diagnostic(ErrorCode.ERR_DefaultLiteralNoTargetType, "default").WithLocation(8, 13),
+                // (9,13): error CS8802: The receiver of a `with` expression must have a valid non-void type.
+                //         _ = null with { };
+                Diagnostic(ErrorCode.ERR_InvalidWithReceiverType, "null").WithLocation(9, 13)
+            );
+        }
+
+        [Fact]
+        public void WithExpr2()
+        {
+            var src = @"
+data class C(int X)
+{
+    public static void Main()
+    {
+        var c = new C(0);
+        c = c with { };
+    }
+}";
+            var comp = CreateCompilation(src);
+            comp.VerifyDiagnostics(
+                // (7,13): error CS8803: The 'with' expression requires the receiver type 'C' to have a single accessible non-inherited instance method named "Clone".
```

Comment:
Not generating clone by default currently? If that's the point of this test, consider using a non-record type, as we'll just have to adjust it when we start generating them. And if the point is to have a place that will change when we do start generating, consider leaving a prototype comment to record that intention.

Reply:
Yup, just didn't want to deal with both parts at the same time since they should be somewhat independent.